client/image_push.go

```diff
@@ -36,6 +37,14 @@ func (cli *Client) ImagePush(ctx context.Context, image string, options image.Pu
 		}
 	}
 
+	if options.Platform != "" {
+		if options.Platform == "local" {
+			query.Set("platform", platforms.DefaultString())
```

Comment:
cc @tianon 

More platform strings :)

Reply:
oof, and generated in client code to pass to the daemon :sob:

local in the client is not local to the daemon :sob:

Reply:
Made `local` mean the platform local to the client for consistency with buildx `--platform local`.


That's why there's also `remote` (although I think I'd prefer to call it `host`) that is handled on the daemon side and defaults to the daemon platform: https://github.com/moby/moby/blob/dfe5a6c6f70c3093be68f7552caaa8bc6b2e9b03/api/server/router/image/image_routes.go#L554-L559
Happy to hear your suggestions if you have better ideas 😄

Reply:
Definitely creeping into ["build" vs "host" vs "target"](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html) confusion territory here -- what's the use case for building for the _client_-local host's platform when talking to a different daemon?

(Some of the context for this discussion is https://github.com/containerd/platforms/pull/6, which actually makes `platforms.DefaultString()` _unsafe_ to pass across process/machine boundaries like this, but I think the issue here is larger.)

Reply:
Probably because I'd mentioned it in an earlier review, but more "the client shouldn't have to know the daemon's platform... also should we do the buildx thing for client platform?"
It may well be not needed.

Reply:
> what's the use case for building for the client-local host's platform when talking to a different daemon?

Personally, `--platform local` was useful to me for extracting binaries out of docker images to use on my host system, like:
```
echo 'FROM moby/moby-bin:master' | docker buildx build  --platform local -o type=local,dest=asdadsad -
```

So, mostly thinking of remote daemon or, a more down-to-earth case: DD on non-Linux platforms - in the above case I really don't care about the VM, I just want to get the image for my host.

However, `local` might be less useful than the `remote/host/daemon` for the interactive case, because most of the time I know what platform my client host is running on. Still, it could be useful in scripts.

Also, in case of a remote Docker host - users might want to push the image for the platform of their host:
```bash
$ docker -c beef tag <some-image> myregistry.local/some-image
$ docker -c beef push --platform local myregistry.local/some-image
$ docker run myregistry.local/some-image
```

But still, I think that's mostly useful for scripts, as in the interactive case it's highly likely that user would pass the platform explicitly anyway.

Reply:
Anyway, I'm fine with dropping these for now. We can always add them later if there's a real demand for it.

Reply:
I would advise against using a single string field to represent the platform in the API. The first issue is highlighted here that it is too tempting to override it with special values. The second issue is that the string format is now a part of the API which it is not intended for. In containerd we only use the string for CLI or config, but the API always uses the structured form (https://github.com/containerd/containerd/blob/main/api/types/platform.proto) or its embedded in json in the original form (https://github.com/opencontainers/image-spec/blob/main/specs-go/v1/descriptor.go#L53). 

The point of this change is to make the API endpoint behavior more explicit which filling in a variable "remote" value doesn't really accomplish. The way the current API behaves is already "--platform remote" and I worry by adding this flag here we end up changing the API to default to only push all platforms and end up recommending users to add "--platform remote". I think its best to keep the default to push whatever what pulled for that same image, so the equivalent of "--platform remote" if only a single platform was pulled or push all platforms if we can support pull all platforms. The case of pushing a single platform which differs from the remote after pulling multiple platforms seems like an edge case and is likely better handled with a separate command to edit an image with a new manifest.

Reply:
Removed the single string and added parameters that correspond to the fields of `ocispec.Platform`.